### ISSUE TITLE
Strongly type parameters of extrinsic api

### ISSUE THREAD

**Issue by clangenb** [leonardocustodio/polkadart#406]
Currently, the outer api of the generated code for the extrinsic omits the type annotation, implicitly making it dynamic.

Example:

```dart

  /// See [`Pallet::transfer_allow_death`].
  _i10.RuntimeCall transferAllowDeath({
    required dest,
    required value,
  }) {
    final _call = _i11.Call.values.transferAllowDeath(
      dest: dest,
      value: value,
    );
    return _i10.RuntimeCall.values.balances(_call);
  }
```

The inner call `_i11.Call.values.transferAllowDeath` has explicit type annotations though, so I guess it should not be too hard to put them also in the outer api too.

It would be beneficial to strongly type the outer api too to detect type errors at compile time.

**Comment by leonardocustodio** [comment]
@clangenb can you give a shot in the PR and check if everything on your side continues to work?

**Comment by clangenb** [comment]
Yes, will do that tomorrow. Thanks for the quick fix!

**Comment by clangenb** [comment]
Integrated it in https://github.com/encointer/encointer-wallet-flutter/pull/1629, and it and works. This gives me some piece of mind. :)

**Comment by leonardocustodio** [comment]
Released under `polkadart_cli 0.3.2`
